File: dev/20190402/k2works/sam-app/HelloWorldFunction/src/main/java/fizzbuzz/FizzBuzzType01.java
package fizzbuzz;

public class FizzBuzzType01 extends IFizzBuzzType {
    public FizzBuzzValue generate(int number) {
        boolean fizz = isFizz(number);
        boolean buzz = isBuzz(number);
        if (fizz && buzz) return new FizzBuzzValue(number,"FizzBuzz");
        if (fizz) return new FizzBuzzValue(number,"Fizz");
        if (buzz) return new FizzBuzzValue(number,"Buzz");
        return new FizzBuzzValue(number,Integer.toString(number));
    }
}

File: dev/20190822/k2works/src/Domain/Type/FizzBuzzType01.ts
import { IFizzBuzzType } from "../../Application/Service/IFizzBuzzType";
import { FizzBuzzValue } from "../Model/FizzBuzzValue";

export class FizzBuzzType01 extends IFizzBuzzType {
  public generate(input: number): FizzBuzzValue {
    const isFizz: boolean = input % 3 === 0;
    const isBuzz: boolean = input % 5 === 0;

    if (isFizz && isBuzz) {
      return new FizzBuzzValue(input, IFizzBuzzType.FIZZ_BUZZ);
    }
    if (isFizz) {
      return new FizzBuzzValue(input, IFizzBuzzType.FIZZ);
    }
    if (isBuzz) {
      return new FizzBuzzValue(input, IFizzBuzzType.BUZZ);
    }

    return new FizzBuzzValue(input, input.toString());
  }
}
